var/spack/repos/builtin/packages/adios2/package.py

```diff
@@ -120,6 +120,7 @@ class Adios2(CMakePackage, CudaPackage, ROCmPackage):
     depends_on("cuda", when="+cuda ~kokkos")
 
     # Kokkos support
+    depends_on("kokkos@3.7:", when="+kokkos ~cuda ~rocm ~sycl")
     depends_on("kokkos@3.7: +cuda +wrapper", when="+kokkos +cuda")
     depends_on("kokkos@3.7: +rocm", when="+kokkos +rocm")
     depends_on("kokkos@3.7: +sycl", when="+kokkos +sycl")
```

Comment:
I'm not sure why you specify `~cuda` etc. in the first directive. I think more readable way to write this may be:
```suggestion
    with when("+kokkos"):
        depends_on("kokkos@3.7:")
        depends_on("kokkos +cuda +wrapper", when="+cuda")
        depends_on("kokkos +rocm", when="+rocm")
        depends_on("kokkos +sycl", when="+sycl")
```